client/src/components/CommentApp/comments.js

```diff
@@ -81,34 +124,73 @@ class FieldLevelCommentWidget {
+    annotationTemplateNode,
+    commentApp
   }) {
     this.fieldNode = fieldNode;
     this.contentpath = getContentPath(fieldNode);
     this.commentAdditionNode = commentAdditionNode;
     this.annotationTemplateNode = annotationTemplateNode;
-    this.commentNumber = 0;
-    this.commentsEnabled = false;
-  }
-  onRegister(makeComment) {
+    this.shown = false;
+    this.commentApp = commentApp;
+  }
+  register() {
+    const { selectEnabled } = this.commentApp.selectors;
+    const initialState = this.commentApp.store.getState();
+    let currentlyEnabled = selectEnabled(initialState);
+    const selectCommentsForContentPath = this.commentApp.utils.selectCommentsForContentPathFactory(
+      this.contentpath
+    );
+    let currentComments = selectCommentsForContentPath(initialState);
+    this.updateVisibility(currentComments.length === 0 && currentlyEnabled);
+    const unsubscribeWidget = this.commentApp.store.subscribe(() => {
+      const state = this.commentApp.store.getState();
+      const newComments = selectCommentsForContentPath(state);
+      const newEnabled = selectEnabled(state);
+      const commentsChanged = (currentComments !== newComments);
+      const enabledChanged = (currentlyEnabled !== newEnabled);
+      if (commentsChanged) {
+        // Add annotations for any new comments
+        currentComments = newComments;
+        currentComments.filter((comment) => comment.annotation === null).forEach((comment) => {
+          const annotation = this.getAnnotationForComment(comment);
+          this.commentApp.updateAnnotation(
+            annotation,
+            comment.localId
+          );
+          annotation.subscribeToUpdates(comment.localId, this.commentApp.store);
+        });
+      }
+      if (enabledChanged || commentsChanged) {
+        // If comments have been enabled or disabled, or the comments have changed
+        // check whether to show the widget (if comments are enabled and there are no existing comments)
+        currentlyEnabled = newEnabled;
+        this.updateVisibility(currentComments.length === 0 && currentlyEnabled);
+      }
+    });
+    initialState.comments.comments.forEach((comment) => {
+      // Add annotations for any comments already in the store
+      if (comment.contentpath === this.contentpath) {
+        const annotation = this.getAnnotationForComment(comment);
+        this.commentApp.updateAnnotation(annotation, comment.localId);
+        annotation.subscribeToUpdates(comment.localId);
+      }
+    });
     this.commentAdditionNode.addEventListener('click', () => {
-      makeComment(this.getAnnotationForComment(), this.contentpath);
+      // Make the widget button clickable to add a comment
+      const annotation = this.getAnnotationForComment();
+      const localId = this.commentApp.makeComment(annotation, this.contentpath);
+      annotation.subscribeToUpdates(localId);
     });
+    return unsubscribeWidget; // TODO: listen for widget deletion and use this
```

Comment:
Should this TODO be addressed before merging?

Reply:
This is only needed within streamfield. As @kaedroho is coming onto the project this week, and we're going to need to integrate with Telepath anyway, I think this is something to address when we integrate into the new streamfield system. The note is really just to make sure we do this before we go live.